### PR TITLE
port to qt5, updated qcustomplot

### PR DESCRIPTION
**Pull request by gagikh**
port to qt5.15, updated cmakefiles.txt, qcustomplot.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,36 +1,44 @@
...
 )
 
-QT4_WRAP_CPP(header_files_moc ${header_files})
+#QT4_WRAP_CPP(header_files_moc ${header_files})
 
-INCLUDE(${QT_USE_FILE})
+#INCLUDE(${QT_USE_FILE})
 
-ADD_DEFINITIONS(${QT_DEFINITIONS})
+#ADD_DEFINITIONS(${QT_DEFINITIONS} -fPIC)
```
Comment (Kev-J) [inline CMakeLists.txt]: Hi @gagikh 
Thanks a lot for your contribution. I think you can remove those commented lines.

File: src/OddSpinBox.hpp [hunk 2]
```diff
@@ -1,40 +1,21 @@
...
- * (at your option) any later version.
- * 
- * PID-autotune is distributed in the hope that it will be useful,
- * but WITHOUT ANY WARRANTY; without even the implied warranty of
- * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
- * GNU General Public License for more details.
- * 
- * You should have received a copy of the GNU General Public License
- * along with PID-autotune.  If not, see <http://www.gnu.org/licenses/>.
- */
```
Comment (Kev-J) [inline src/OddSpinBox.hpp:20]: Could you please keep the header?

File: src/OddSpinBox.hpp [hunk 3]
```diff
@@ -1,40 +1,21 @@
...
 
 #include <QSpinBox>
+#include <QtCore/QObject>
 
 // Thanks to ChrisW67 from the Qtproject forum: 
 // http://qt-project.org/forums/viewthread/24037
 class OddSpinBox : public QSpinBox {
-    Q_OBJECT;
-    public:
+    // Q_OBJECT
```
Comment (Kev-J) [inline src/OddSpinBox.hpp]: You can remove the comment

File: src/OddSpinBox.hpp [hunk 4]
```diff
@@ -1,40 +1,21 @@
...
+public:
     OddSpinBox(QWidget *parent = 0)
         : QSpinBox(parent)
     {
         setSingleStep(2);
     }
 
+	//virtual ~OddSpinBox() = default;
```
Comment (Kev-J) [inline src/OddSpinBox.hpp]: Same here

**Comment by gagikh** [comment]
Updated notes from Kev-J.

**Pull request merged**